File: algorithm/src/main/java/baekjoon/p10988/Main.java
package baekjoon.p10988;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println(solve(sc.next()));
    }

    private static int solve(String input) {
        int len = input.length();

        for (int i = 0, j = len - 1; i <= j; i++, j--) {
            if (input.charAt(i) != input.charAt(j)) {
                return 0;
            }
        }

        return 1;
    }

}

File: algorithm/src/main/java/baekjoon/p10988/Main.kt
package baekjoon.p10988

fun main(args: Array<String>) {
    val input = readLine().toString()
    println(solve(input))
}

private fun solve(input: String): Int {
    val len = input.length

    for (i in 0 until len / 2) {
        if (input[i] != input[len - 1 - i]) {
            return 0
        }
    }

    return 1
}